src/test/java/org/jenkinsci/plugins/github_branch_source/EventsTest.java

```diff
@@ -0,0 +1,174 @@
+        ghEvent = callOnEvent(subscriber, "EventsTest/pushEventUpdated.json");
+        waitAndAssertReceived(true);
+    }
+
+    @Test
+    public void pullRequestEvents() throws Exception {
+        PullRequestGHEventSubscriber subscriber = new PullRequestGHEventSubscriber();
+
+        firedEventType = SCMEvent.Type.CREATED;
+        ghEvent = callOnEvent(subscriber, "EventsTest/pullRequestEventCreated.json");
+        waitAndAssertReceived(true);
+
+        firedEventType = SCMEvent.Type.REMOVED;
+        ghEvent = callOnEvent(subscriber, "EventsTest/pullRequestEventRemoved.json");
+        waitAndAssertReceived(true);
+
+        firedEventType = SCMEvent.Type.UPDATED;
+        ghEvent = callOnEvent(subscriber, "EventsTest/pullRequestEventUpdated.json");
+        waitAndAssertReceived(true);
+
+        ghEvent = callOnEvent(subscriber, "EventsTest/pullRequestEventUpdatedSync.json");
+        waitAndAssertReceived(true);
+    }
+
+    @Test
+    public void repositoryEvents() throws Exception {
+        GitHubRepositoryEventSubscriber subscriber = new GitHubRepositoryEventSubscriber();
+
+        firedEventType = SCMEvent.Type.CREATED;
+        ghEvent = callOnEvent(subscriber, "EventsTest/repositoryEventCreated.json");
+        waitAndAssertReceived(true);
+
+        ghEvent = callOnEvent(subscriber, "EventsTest/repositoryEventNotFiredNotFork.json");
+        waitAndAssertReceived(false);
+
+        ghEvent = callOnEvent(subscriber, "EventsTest/repositoryEventNotFiredWrongAction.json");
+        waitAndAssertReceived(false);
+    }
+
+    private GHSubscriberEvent callOnEvent(PushGHEventSubscriber subscriber, String eventPayloadFile) throws IOException {
+        GHSubscriberEvent event = createMockEvent(eventPayloadFile);
+        subscriber.onEvent(event);
+        return event;
+    }
+
+    private GHSubscriberEvent callOnEvent(PullRequestGHEventSubscriber subscriber, String eventPayloadFile) throws IOException {
+        GHSubscriberEvent event = createMockEvent(eventPayloadFile);
+        subscriber.onEvent(event);
+        return event;
+    }
+
+    private GHSubscriberEvent callOnEvent(GitHubRepositoryEventSubscriber subscriber, String eventPayloadFile) throws IOException {
+        GHSubscriberEvent event = createMockEvent(eventPayloadFile);
+        subscriber.onEvent(event);
+        return event;
+    }
+
+    private GHSubscriberEvent createMockEvent(String eventPayloadFile) throws IOException {
+        GHSubscriberEvent event = mock(GHSubscriberEvent.class);
+        when(event.getPayload()).thenReturn(IOUtils.toString(getClass().getResourceAsStream(eventPayloadFile)));
+        when(event.getGHEvent()).thenReturn(null);
+        when(event.getOrigin()).thenReturn("myOrigin");
+        when(event.getTimestamp()).thenReturn(123456789L);
+
+        return event;
+    }
+
+    private void waitAndAssertReceived(boolean received) throws InterruptedException {
+        Thread.sleep(5100);
```

Comment:
Oh dear me. No. Please see `SCMEvents.getWatermark()` and `SCMEvents.awaitAll(watermark)` in the scm-api tests artifact